server/testutil/testbazel/testbazel.go

```diff
@@ -2,62 +2,142 @@ package testbazel
 
 import (
 	"context"
+	"fmt"
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strings"
 	"sync"
 	"testing"
 	"time"
 
+	"github.com/bazelbuild/rules_go/go/runfiles"
 	"github.com/buildbuddy-io/buildbuddy/server/testutil/testfs"
 	"github.com/buildbuddy-io/buildbuddy/server/util/bazel"
 	"github.com/stretchr/testify/require"
-
-	"github.com/bazelbuild/rules_go/go/runfiles"
 )
 
 var (
 	// Version is the bazel version of the embedded test bazel binary.
 	//
 	// Injected via x_defs.
 	Version string
-	// BazelBinaryPath specifies the path to the bazel binary used for
-	// invocations. Must match the path in the build rule.
+
+	// bazelRlocationpath specifies the path to the bazel binary used for
+	// invocations.
+	//
+	// This binary should not be run as-is, since it doesn't have the
+	// pre-configured install_base which is required to avoid extracting the
+	// installation during tests, and it doesn't have the pre-warmed repository
+	// cache which is required to avoid making network requests.
 	//
 	// Injected via x_defs.
-	BazelBinaryPath string
-	// InstallBasePath is the path to the pre-extracted Bazel installation
+	bazelRlocationpath string
+
+	// outdirRlocationpath contains the pre-extracted Bazel installation
 	// relative to the runfiles dir.
 	//
 	// Injected via x_defs.
-	installBasePath string
+	outdirRlocationpath string
 
 	initOnce sync.Once
 )
 
-// BinaryPath returns the path to the bazel binary.
-func BinaryPath(t *testing.T) string {
-	// Write an entrypoint script that runs bazel with --install_base.
+// InitWorkspace creates a MODULE.bazel file and MODULE.bazel.lock file in the
+// given workspace dir if they don't already exist. The lockfile that is written
+// is sufficient for bazel to not make any network requests when using built-in
+// rules such as genrule, sh_binary, and sh_test.
+func InitWorkspace(t testing.TB, workspaceDir string) {
```

Comment:
nit: rename to InitModule?

Reply:
Good call - also renamed `MakeTempWorkspace` to `MakeTempModule`